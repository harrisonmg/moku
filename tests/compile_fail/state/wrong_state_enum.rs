use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod machine {}

    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
    enum BlonkyState {}
    impl StateEnum for BlonkyState {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    #[superstate(Top)]
    impl State<BlonkyState> for Bottom {}
}

fn main() {}
