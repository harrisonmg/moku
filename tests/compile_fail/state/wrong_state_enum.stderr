error: implementations of moku::State in this module must use only BlinkyState as the generic
  --> tests/compile_fail/state/wrong_state_enum.rs:21:5
   |
21 |     #[superstate(Top)]
   |     ^
